cmd/bosun/sched/notify.go

```diff
@@ -224,3 +225,85 @@ func (s *Schedule) AddNotification(ak expr.AlertKey, n *conf.Notification, start
 	}
 	s.Notifications[ak][n.Name] = started
 }
+
+var actionNotificationSubjectTemplate *ttemplate.Template
+var actionNotificationBodyTemplate *htemplate.Template
+
+func init() {
+	subject := `{{$first := index .States 0}}{{$count := len .States}}
+{{.User}} {{.ActionType}}
+{{if gt $count 1}} {{$count}} Alerts. 
+{{else}} Incident #{{$first.Last.IncidentId}} ({{$first.Subject}}) 
+{{end}}`
+	body := `{{$count := len .States}}{{.User}} {{.ActionType}} {{$count}} alert{{if gt $count 1}}s{{end}}: <br/>
+<strong>Message:</strong> {{.Message}} <br/>
+<strong>Incidents:</strong> <br/>
+<ul>
+	{{range .States}}
+		<li>
+			<a href="{{$.IncidentLink .Last.IncidentId}}">#{{.Last.IncidentId}}:<a/> 
+			{{.Subject}}
+		</li>
+	{{end}}
+</ul>`
+	actionNotificationSubjectTemplate = ttemplate.Must(ttemplate.New("").Parse(strings.Replace(subject, "\n", "", -1)))
+	actionNotificationBodyTemplate = htemplate.Must(htemplate.New("").Parse(body))
+}
+
+func (s *Schedule) ActionNotify(at ActionType, user, message string, aks []expr.AlertKey) {
+	groupings := s.groupActionNotifications(aks)
+
+	for notification, aks := range groupings {
+		incidents := []*State{}
+		for _, ak := range aks {
+			state := s.GetStatus(ak)
```

Comment:
What if `groupActionNotifications` returned a `map[*conf.Notification][]*State`? This would prevent calling GetStatus twice for each ak.